rotkehlchen/tests/api/test_ethereum_transactions.py

```diff
@@ -1051,3 +1056,108 @@ def query_transactions(rotki, tx_module: EthTransactions) -> None:
+        - Filtering by protocol (counterparty)
+        - Filtering by both asset and a protocol
+        - Transaction has multiple related events
+        - Transaction has no related events
+        - Multiple transactions are queried
+    """
+    logging.getLogger('rotkehlchen.externalapis.etherscan').disabled = True
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+    db = rotki.data.db
+    tx1 = create_tx(tx_hash=b'1')
+    tx2 = create_tx(tx_hash=b'2')
+    tx3 = create_tx(tx_hash=b'3')
+    event1 = create_tx_event(tx_hash=b'1', index=1, asset=A_ETH)
+    event2 = create_tx_event(tx_hash=b'1', index=2, asset=A_ETH, counterparty='EXAMPLE_PROTOCOL')
+    event3 = create_tx_event(tx_hash=b'1', index=3, asset=A_BTC, counterparty='EXAMPLE_PROTOCOL')
+    event4 = create_tx_event(tx_hash=b'2', index=4, asset=A_BTC)
+    dbethtx = DBEthTx(db)
+    dbethtx.add_ethereum_transactions([tx1, tx2], relevant_address=ethereum_accounts[0])
+    dbethtx.add_ethereum_transactions([tx3], relevant_address=ethereum_accounts[1])
+    dbevents = DBHistoryEvents(db)
+    dbevents.add_history_events([event1, event2, event3, event4])
+
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            'ethereumtransactionsresource',
+        ),
+        json={
+            'asset': A_BTC.serialize(),
+            'address': ethereum_accounts[0],
+        },
+    )
+    result = assert_proper_response_with_result(response)
+    expected = generate_tx_entries_response([(tx1, [event3]), (tx2, [event4])])
+    assert result['entries'] == expected
+
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            'ethereumtransactionsresource',
+        ),
+        json={'asset': A_ETH.serialize()},
+    )
+    result = assert_proper_response_with_result(response)
+    expected = generate_tx_entries_response([(tx1, [event1, event2])])
+    assert result['entries'] == expected
+
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            'ethereumtransactionsresource',
+        ),
+        json={'asset': A_BTC.serialize()},
+    )
+    result = assert_proper_response_with_result(response)
+    expected = generate_tx_entries_response([(tx1, [event3]), (tx2, [event4])])
+    assert result['entries'] == expected
+
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            'ethereumtransactionsresource',
+        ),
+        json={'protocols': ['EXAMPLE_PROTOCOL']},
+    )
+    result = assert_proper_response_with_result(response)
+    expected = generate_tx_entries_response([(tx1, [event2, event3])])
+    assert result['entries'] == expected
+
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server,
+            'ethereumtransactionsresource',
+        ),
+        json={
+            'asset': A_BTC.serialize(),
+            'protocols': ['EXAMPLE_PROTOCOL'],
```

Comment:
Also add a test for frontend giving to the backend an empty list of protocols and see that the backend reacts properly. I would even say empty list of protocols makes no sense. So reject it at the api level please.

Reply:
Aha. Alright, will add logic to reject it

Reply:
Added